### DISCUSSION TITLE
You may need an appropriate loader to handle this file type (importing `@walletconnect/sign-client 2.7.8`)

### DISCUSSION THREAD

**Discussion by KatnessChen** [WalletConnect/walletconnect-monorepo#3397] [Developer Support]
**Describe the bug**

Hello everyone, I encounter plugin import issue with my local environment. Appreciate any advices for trouble shooting!

The plugin I intended to use: https://docs.walletconnect.com/2.0/web/sign/dapp-usage

1. Import `@walletconnect/sign-client` & `@walletconnect/modal`

then run the frontend project, got warnings:

```
error  in ./node_modules/@walletconnect/sign-client/dist/index.es.js
Module parse failed: Unexpected token (1:4244)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file.
```

2. add loader to webpack config
```
{
  test: /node_modules[\\/]@walletconnect/,
  use: [
    {
      loader: 'babel-loader',
    },
  ],
},
```

3. error message changes to 
```
./node_modules/@walletconnect/core/dist/index.es.js 6362:404-406
"export 'default' (imported as 'Bt') was not found in '@walletconnect/keyvaluestorage'

./node_modules/@walletconnect/utils/dist/index.es.js 388:9-11
"export 'getWindowMetadata' (imported as 'vn') was not found in '@walletconnect/window-metadata'

react-dom.development.js:13035 Uncaught ReferenceError: exports is not defined
    at eval (index.js:13:23)
    at ./node_modules/@walletconnect/keyvaluestorage/dist/cjs/browser/index.js 
```

Screenshot
![Screenshot 2023-06-14 at 15 24 36](https://github.com/WalletConnect/walletconnect-monorepo/assets/34178240/77f1e9e6-8cc4-45f2-b6b9-c474a1e3eaba)

4. checked the source code `@walletconnect/sign-client/dist/index.es.js#1:4244`  and the error happens at a `??` character

so I added `@babel/plugin-transform-nullish-coalescing-operator` plugin to webpack. But seems not helping.

![Screenshot 2023-06-14 at 14 51 45](https://github.com/WalletConnect/walletconnect-monorepo/assets/34178240/9250ebd1-08fd-4c47-9862-dc113e02a3ff)

**SDK Version (if relevant)**
- @walletconnect/modal: ^2.4.5,
- @walletconnect/sign-client": ^2.8.0,

**To Reproduce**
Please refer to the "Describe the bug" section

**Expected behavior**
WalletConnect able to up and running on browser

**Screenshots**
See above

**Desktop (please complete the following information):**
 - OS: macOS Ventura Version 13.4 (22F66)
 - Browser: Chrome
 - Version: 114

**Additional context**
package.json
```
dependencies: {
 "react": "^16.13.1",
}
devDependencies: {
 "tslint": "^5.20.0",
 "typescript": "^3.7.2",   
 "@babel/core": "^7.6.4",
 "eslint": "6.8.0",
},
"engines": {
  "node": "12.20.0"
},
```
webpack config
```
const config = {
  module: {
    rules: [
      {
        test: /node_modules[\\/]@walletconnect/,
        use: [
          {
            loader: 'babel-loader',
          },
        ],
      },
      {
        test: /\.(ts|tsx|js)$/,
        exclude: [/(node_modules)/],
        use: [
          {
            loader: 'babel-loader',
            options: {
              cacheDirectory: true,
            },
          },
        ],
      },
     // ....
    ],
  },
}
```

**Comment by outlawcode** [comment]
I got same error!

**Comment by glitch-txs** [comment]
Try the CDN packages instead or to upgrade your bundler configuration 🙏
